submissions/Ant-C-tech/friends-app/js/app.js

```diff
@@ -0,0 +1,322 @@
+            break
+        case 'lastName_a-z':
+            CURRENT_FRIENDS.sort(byField('name', 'last'))
+            break
+        case 'lastName_z-a':
+            CURRENT_FRIENDS.sort(byField('name', 'last')).reverse()
+            break
+
+        default:
+            break
+    }
+
+    changeContent(createFriendsScreen(CURRENT_FRIENDS), SHOW_ELEM_PRIMARY_ANIMATION, HIDE_ELEM_PRIMARY_ANIMATION)
+}
+
+function resetFilter() {
+    //In case user enter to app through sidenav (not using startButton)
+    if (!IS_MUSIC_STOPPED_BY_USER) {
+        playMusic()
+    }
+    showSearchBar()
+    closeSideNav()
+
+    CURRENT_FRIENDS = []
+    if (document.querySelector("input[type='radio'][name='gender']:checked")) {
+        document.querySelector("input[type='radio'][name='gender']:checked").checked = false
+    }
+    if (document.querySelector("input[type='radio'][name='sort']:checked")) {
+        document.querySelector("input[type='radio'][name='sort']:checked").checked = false
+    }
+    changeContent(createFriendsScreen(FRIENDS_SOURCE), SHOW_ELEM_PRIMARY_ANIMATION, HIDE_ELEM_PRIMARY_ANIMATION)
+}
+
+function closeSideNav() {
+    SEARCH_INPUT.value = ''
+    document.querySelector('.search__hint').classList.remove('active')
+}
+
+function activateSideNav() {
+    const sideNavIco = document.querySelectorAll('.sidenav')
+    M.Sidenav.init(sideNavIco, {})
+}
+
+function playMusic() {
+    APP_AUDIO.play()
+    MUSIC_BUTTON.classList.add('musicButton-active')
+    MUSIC_BUTTON.classList.add('musicButton-animation')
+    MUSIC_BUTTON.addEventListener('click', stopMusic, { once: true })
+}
+
+function stopMusic() {
+    APP_AUDIO.pause()
+    MUSIC_BUTTON.classList.remove('musicButton-animation')
+    MUSIC_BUTTON.addEventListener('click', playMusic, { once: true })
+    IS_MUSIC_STOPPED_BY_USER = true
+}
+
+function getRandomIntInclusive(min, max) {
+    min = Math.ceil(min)
+    max = Math.floor(max)
+    return Math.floor(Math.random() * (max - min + 1) + min)
+}
+
+function byField(fieldName, subFieldName) {
+    return (a, b) => {
+        if (a[fieldName][subFieldName] > b[fieldName][subFieldName]) {
+            return 1
+        } else if (a[fieldName][subFieldName] < b[fieldName][subFieldName]) {
+            return -1
+        } else {
+            return 0
+        }
+    }
+}
+
+function handleErrors(response) {
+    if (!response.ok) {
+        throw Error(response.statusText)
+    }
+    return response
+}
+
+function showCurrentContent() {
+    MAIN.innerHTML = ''
+    MAIN.innerHTML = CURRENT_CONTENT
+    MAIN.classList.remove(CURRENT_ANIMATION_HIDE)
+    MAIN.classList.add(CURRENT_ANIMATION_SHOW)
+}
+
+function clearAfterAnimation() {
+    MAIN.classList.remove(CURRENT_ANIMATION_SHOW)
+    MAIN.classList.remove(CURRENT_ANIMATION_SPEED)
+    MAIN.classList.add('scroll')
+}
+
+function raf(fn) {
+    window.requestAnimationFrame(function() {
+        window.requestAnimationFrame(function() {
```

Comment:
How calling one rAf inside another rAF solves your issue and what issue you have that you used such a solution?

Reply:
I am sorry. I was making "Website Performance Optimization" task and it was just bad idea to conduct experiments with "Friends App". I forgot to delete this one. It was my mistake.